test/server/http/admin_test.cc

```diff
@@ -233,69 +234,79 @@ TEST_P(AdminInstanceTest, Runtime) {
   Http::HeaderMapImpl header_map;
   Buffer::OwnedImpl response;
 
-  std::unordered_map<std::string, const Runtime::Snapshot::Entry> entries{
-      {"string_key", {"foo", {}}}, {"int_key", {"1", {1}}}, {"other_key", {"bar", {}}}};
   Runtime::MockSnapshot snapshot;
   Runtime::MockLoader loader;
+  auto layer1 = std::make_unique<NiceMock<Runtime::MockOverrideLayer>>();
+  auto layer2 = std::make_unique<NiceMock<Runtime::MockOverrideLayer>>();
+  std::unordered_map<std::string, Runtime::Snapshot::Entry> entries1{
+      {"string_key", {"foo", {}}}, {"int_key", {"1", {1}}}, {"other_key", {"bar", {}}}};
+  std::unordered_map<std::string, Runtime::Snapshot::Entry> entries2{
+      {"string_key", {"override", {}}}, {"extra_key", {"bar", {}}}};
+
+  ON_CALL(*layer1, name()).WillByDefault(testing::ReturnRefOfCopy(std::string{"layer1"}));
+  ON_CALL(*layer1, values()).WillByDefault(testing::ReturnRef(entries1));
+  ON_CALL(*layer2, name()).WillByDefault(testing::ReturnRefOfCopy(std::string{"layer2"}));
+  ON_CALL(*layer2, values()).WillByDefault(testing::ReturnRef(entries2));
+
+  std::vector<Runtime::Snapshot::OverrideLayerConstPtr> layers;
+  layers.push_back(std::move(layer1));
+  layers.push_back(std::move(layer2));
+  EXPECT_CALL(snapshot, getLayers()).WillRepeatedly(testing::ReturnRef(layers));
+
+  const std::string expected_json = R"EOF({
+    "layers": [
+        "layer1",
+        "layer2"
+    ],
+    "entries": {
+        "extra_key": [
```

Comment:
Can we make each entry look like:

```
"extra_key": {
  "final_value": "...",
  "layers": [ ... ]
}
```
I think it would make it easier to see what is the current value quickly.